src/ride-request/ride-request.controller.ts

```
@@ -0,0 +1,89 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  ParseUUIDPipe,
+  Patch,
+  Post,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { RideRequestService } from './ride-request.service';
+import { AuthGuardService } from '@/auth-guard/auth-guard.service';
+import { CreateRideRequestData } from './dto/create-ride-request-data';
+import type { RequestWithUser } from '@/types/RequestWithUser';
+import { UpdateRideRequestData } from './dto/update-ride-request-data';
+
+@Controller('ride-requests')
+export class RideRequestController {
+  constructor(private readonly rideRequestService: RideRequestService) {}
+
+  @UseGuards(AuthGuardService)
+  @HttpCode(HttpStatus.CREATED)
+  @Post()
+  async createRideRequest(
+    @Body() body: CreateRideRequestData,
+    @Req() request: RequestWithUser,
+  ) {
+    const userId = request.decodedData.id;
```

Maybe we could check and throw NotFoundErrors for userId not found

---

I have handled this in guard service. so i did not check it here.